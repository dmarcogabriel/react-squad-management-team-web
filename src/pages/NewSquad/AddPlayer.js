import React from 'react';
import { MdAdd } from 'react-icons/md';

const AddPlayer = () => {
  return (
    <div className="rounded-full border-dashed border-2 p-5 text-white">
      <MdAdd size={22} />
    </div>
  );
};

export default AddPlayer;
