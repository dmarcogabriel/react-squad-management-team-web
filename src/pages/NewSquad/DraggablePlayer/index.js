import React from 'react';
import { useDrag } from 'react-dnd';
import cn from 'classnames';

const DraggablePlayer = ({ player }) => {
  const [{ isDragging }, drag] = useDrag({
    item: { type: 'player', id: player.player_id },
    collect: (monitor) => ({
      isDragging: !!monitor.isDragging(),
    }),
  });

  return (
    <div
      ref={drag}
      className={cn(
        'bg-gradient-to-t from-gray-400 to-white',
        'border-dashed border-2 rounded-sm p-3 my-2 cursor-move',
        isDragging && 'opacity-50'
      )}
    >
      <div className="flex flex-col md:flex-row justify-between">
        <p className="text-base font-bold">
          Name:
          <span className="text-primary-dark">{` ${player.fullname}`}</span>
        </p>
        <p className="text-base font-bold">
          Age:
          <span className="text-primary-dark">{` ${player.age}`}</span>
        </p>
      </div>
      <p className="text-base font-bold">
        Nacionality:
        <span className="text-primary-dark">{` ${player.nationality}`}</span>
      </p>
    </div>
  );
};

export default DraggablePlayer;
