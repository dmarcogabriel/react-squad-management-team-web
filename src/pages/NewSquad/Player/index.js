import React from 'react';
import { MdAdd } from 'react-icons/md';
import cn from 'classnames';
import { useDrop } from 'react-dnd';
import { showPlayerInitials } from '../../../utils/playersInitials';

const Player = ({ className, player, onDropPlayer }) => {
  const [{ isOver }, drop] = useDrop({
    accept: 'player',
    drop: (e) => onDropPlayer(e.id),
    collect: (monitor) => ({
      isOver: !!monitor.isOver(),
    }),
  });

  const hasPlayer = () =>
    player.player
      ? 'bg-opacity-100 text-primary-dark'
      : 'bg-opacity-50 text-white border-2';

  return (
    <div
      data-testid="drop-player"
      ref={drop}
      className={cn(
        'relative rounded-full border-dashed p-8',
        'bg-white inline-block text-white',
        isOver && 'bg-opacity-100',
        hasPlayer(),
        className
      )}
    >
      <div
        className={cn(
          'absolute top-0 left-0',
          'flex items-center justify-center w-full h-full'
        )}
      >
        {player.player ? (
          <p className="text-base">
            {showPlayerInitials(player.player.display_name)}
          </p>
        ) : (
          <MdAdd size={22} />
        )}
      </div>
    </div>
  );
};

export default Player;
