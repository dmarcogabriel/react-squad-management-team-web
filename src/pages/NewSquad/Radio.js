import React from 'react';
import cn from 'classnames';

const Radio = ({
  className = '',
  label,
  options = [],
  onChange = () => {},
}) => {
  const isSelected = (selected) =>
    selected ? 'border-primary-dark' : 'border-gray-500';

  return (
    <div className={className}>
      <p className="font-bold text-base mb-2">{label}</p>

      <div className="flex justify-between">
        {options.map((option) => (
          <button
            type="button"
            key={option.id}
            className="flex items-center"
            onClick={() => onChange(option)}
          >
            <div
              className={cn(
                'rounded-full p-3 border-1 mr-3 relative',
                isSelected(option.selected)
              )}
            >
              {option.selected && (
                <div className="bg-primary-dark w-5 h-5 absolute m-auto top-0 bottom-0 right-0 left-0 rounded-full" />
              )}
            </div>

            <p
              className={cn(
                'inline text-sm',
                option.selected ? 'text-primary-dark' : 'text-gray-500'
              )}
            >
              {option.label}
            </p>
          </button>
        ))}
      </div>
    </div>
  );
};

export default Radio;
