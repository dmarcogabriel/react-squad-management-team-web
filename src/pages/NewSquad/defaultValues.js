export const FORMATION_OPTIONS = [
  {
    id: '3223',
    label: '3-2-2-3',
    selected: false,
  },
  {
    id: '3231',
    label: '3-2-3-1',
    selected: false,
  },
  {
    id: '343',
    label: '3-4-3',
    selected: false,
  },
  {
    id: '352',
    label: '3-5-2',
    selected: false,
  },
  {
    id: '4231',
    label: '4-2-3-1',
    selected: false,
  },
  {
    id: '432',
    label: '4-3-2',
    selected: false,
  },
  {
    id: '442',
    label: '4-4-2',
    selected: false,
  },
  {
    id: '451',
    label: '4-5-1',
    selected: false,
  },
  {
    id: '541',
    label: '5-4-1',
    selected: false,
  },
];

export const TYPE_OPTIONS = [
  { id: 'real', label: 'Real', selected: false },
  { id: 'fantasy', label: 'Fantasy', selected: false },
];
