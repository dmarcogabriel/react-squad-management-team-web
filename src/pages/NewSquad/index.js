import React, { useState } from 'react';
import { useFormik } from 'formik';
import { object, string } from 'yup';
import { MdClose } from 'react-icons/md';
import Card from '../../common/Card';
import Page from '../../common/Page';
import TextInput from './TextInput';
import AddPlayer from './AddPlayer';

const NewSquad = () => {
  const [tags, setTags] = useState([]);

  const deleteTags = (tagName) => {
    setTags(tags.filter((tag) => tag !== tagName));
  };

  const { values, errors, handleChange, handleSubmit } = useFormik({
    initialValues: {
      name: '',
      description: '',
      webSite: '',
      type: '',
      tags: '',
    },
    validationSchema: () =>
      object().shape({
        name: string().required(),
        webSite: string().url(),
        type: string().required(),
      }),
    onSubmit: () => {
      // eslint-disable-next-line
      console.log('Enviou heim');
    },
  });

  return (
    <Page>
      <Card title="Create your team">
        <div>
          <h1 className="text-center text-gray-600 text-base mb-5">
            TEAM INFORMATION
          </h1>

          <div className="flex justify-center">
            <div>
              <TextInput
                label="Team Name"
                id="teamName"
                value={values.name}
                error={errors.name}
                onChange={handleChange('name')}
              />

              <div>
                <label htmlFor="description">
                  <p className="font-bold text-base">Description</p>

                  <textarea
                    id="description"
                    className="w-full h-full p-1 border-2 border-gray-400 rounded"
                    rows="5"
                    value={values.description}
                    onChange={(e) =>
                      handleChange('description')(e.target.value)
                    }
                  />
                </label>
              </div>
            </div>

            <div>
              <TextInput
                label="Team website"
                id="teamWebSite"
                type="text"
                placeholder="https://myteam.com"
                value={values.webSite}
                error={errors.webSite}
                onChange={handleChange('webSite')}
              />

              <div>
                <p className="font-bold text-base">Team Type</p>
                <label htmlFor="teamTypeReal">
                  <input type="radio" />
                  <p className="inline text-sm">Real</p>
                </label>

                <label htmlFor="teamTypeFantasy">
                  <input type="radio" />
                  <p className="inline text-sm text-primary-dark">Fantasy</p>
                </label>
              </div>

              <div>
                <label htmlFor="tags">
                  <p className="font-bold text-base">Tags</p>

                  <div className="rounded border-2 border-gray-400">
                    <div className="flex">
                      {tags.map((tag) => (
                        <button
                          key={tag}
                          type="button"
                          className="flex justify-between items-center py-1 px-2 bg-primary-dark text-white text-sm rounded-full mx-2"
                          onClick={() => deleteTags(tag)}
                        >
                          <p className="inline mr-2">{tag}</p>

                          <MdClose />
                        </button>
                      ))}
                    </div>

                    <textarea
                      id="tags"
                      className="w-full h-full "
                      rows="3"
                      value={values.tags}
                      onKeyDown={(e) => {
                        // todo: 3.1.6 utilizar key não está dando bom
                        if (e.keyCode === 13 || e.keyCode === 186) {
                          setTags([...tags, values.tags]);
                          handleChange('tags')('');
                        } else {
                          handleChange('tags')(values.tags + e.key);
                        }
                      }}
                    />
                  </div>
                </label>
              </div>
            </div>
          </div>
        </div>

        <div>
          <h1 className="text-center text-gray-600 text-base">
            CONFIGURE SQUAD
          </h1>

          <div className="flex">
            <div>
              <div>
                <label htmlFor="formation">
                  <p className="inline text-base font-bold">Formation</p>

                  <select id="formation">
                    <option value="test">3-4-3</option>
                  </select>
                </label>

                <div className="bg-gradient-to-t flex items-center from-secondary-dark to-secondary-light rounded-sm p-10 mb-5">
                  {Array.from(Array(11)).map((_, i) => (
                    <AddPlayer key={String(i)} />
                  ))}
                </div>
              </div>

              <button
                type="button"
                className="bg-gradient-to-t from-secondary-dark to-secondary-light w-full text-white text-base font-bold rounded-sm py-2"
                onClick={handleSubmit}
              >
                Save
              </button>
            </div>

            <div>
              <div>
                <TextInput label="Search Players" id="searchPlayers" />
              </div>

              <div>
                {Array.from(Array(3)).map((_, i) => (
                  <div
                    key={String(i)}
                    className="bg-gradient-to-t from-gray-400 to-white border-dashed border-2 rounded-sm"
                  >
                    <div>
                      <p className="text-base font-bold">
                        Name:
                        <span className="text-primary-dark">
                          {` ${'Cristiano Ronaldo'}`}
                        </span>
                      </p>
                      <p className="text-base font-bold">
                        Age:
                        <span className="text-primary-dark">{` ${'37'}`}</span>
                      </p>
                    </div>
                    <p className="text-base font-bold">
                      Nacionality:
                      <span className="text-primary-dark">{` ${'Portugal'}`}</span>
                    </p>
                  </div>
                ))}
              </div>
            </div>
          </div>
        </div>
      </Card>
    </Page>
  );
};

export default NewSquad;
