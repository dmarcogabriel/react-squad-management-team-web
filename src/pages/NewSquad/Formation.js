import React, { useState, useEffect } from 'react';
import Player from './Player';

const Formation = ({ formation = [], onDropPlayer = () => {} }) => {
  const [rows, setRows] = useState([]);

  useEffect(() => {
    setRows(formation);
  }, [formation]);

  return (
    <div className="bg-gradient-to-t from-secondary-dark to-secondary-light rounded-sm p-2 mb-5 justify-center">
      {rows.map(({ players, rowNumber, id }, i) => (
        <div
          className={`grid grid-cols-3 grid-rows-${rowNumber > 3 ? '2' : '1'}`}
          key={`r${String(i)}`}
        >
          {players.map((player) => (
            <div key={player.position} className="m-2 text-center">
              <Player
                player={player}
                onDropPlayer={(playerId) =>
                  onDropPlayer(playerId, id, player.position)
                }
              />
            </div>
          ))}
        </div>
      ))}
    </div>
  );
};

export default Formation;
