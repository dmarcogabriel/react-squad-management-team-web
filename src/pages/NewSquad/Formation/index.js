import React from 'react';
import cn from 'classnames';
import Player from '../Player';

const Formation = ({ formation = [], onDropPlayer }) => {
  return (
    <div
      data-testid="formation"
      className={cn(
        'bg-gradient-to-t from-secondary-dark to-secondary-light',
        'rounded-sm p-2 mb-5 justify-center'
      )}
    >
      {formation.map(({ players, rowNumber, id }, i) => (
        <div
          data-testid={`formation-row-${id}`}
          className={`grid grid-cols-3 grid-rows-${rowNumber > 3 ? '2' : '1'}`}
          key={`r${String(i)}`}
        >
          {players.map((player) => (
            <div
              data-testid={`formation-player-${player.position}`}
              key={player.position}
              className="m-2 text-center"
            >
              <Player
                player={player}
                onDropPlayer={(playerId) =>
                  onDropPlayer(playerId, id, player.position)
                }
              />
            </div>
          ))}
        </div>
      ))}
    </div>
  );
};

export default Formation;
