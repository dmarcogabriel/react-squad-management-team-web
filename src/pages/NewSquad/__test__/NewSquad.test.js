import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NewSquad from '..';

jest.mock('axios');

describe('<NewSquad />', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<NewSquad location={{ search: '' }} />, div);
  });

  it('should matches snapshot', () => {
    expect(
      renderer.create(<NewSquad location={{ search: '' }} />).toJSON()
    ).toMatchSnapshot();
  });
});
