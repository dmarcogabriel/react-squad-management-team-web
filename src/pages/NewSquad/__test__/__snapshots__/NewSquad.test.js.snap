// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NewSquad /> should matches snapshot 1`] = `
Array [
  <nav
    className="flex px-8 py-2 items-center justify-between bg-gradient-to-r from-primary-dark to-secondary-dark text-white"
    data-testid="header"
  >
    <div>
      <p
        className="text-lg"
      >
        Squad Management Tool
      </p>
    </div>
    <div
      className="flex items-center"
    >
      <p
        className="text-sm mr-2 invisible md:visible"
      >
        John Doe
      </p>
      <p
        className="bg-white py-2 px-3 rounded-full text-black"
      >
        JD
      </p>
    </div>
  </nav>,
  <div
    className="flex flex-col md:flex-row bg-background justify-center w-full px-5 md:px-10"
    data-testid="page"
  >
    <div
      className="bg-white rounded-lg shadow-lg my-10 w-full"
      data-testid="card"
    >
      <div
        className="flex p-8 justify-between items-center border-b-2 border-gray-200"
      >
        <h1
          className="text-secondary-dark font-bold text-md"
          data-testid="card-title"
        >
          Create your team
        </h1>
      </div>
      <div
        className="py-5 px-2"
        data-testid="card-children"
      >
        <div>
          <h1
            className="text-center text-gray-600 text-base mb-5"
          >
            TEAM INFORMATION
          </h1>
          <div
            className="flex flex-col md:flex-row justify-center py-5 md:px-20"
          >
            <div
              className="md:w-full md:px-20"
            >
              <div
                className="w-full"
              >
                <label
                  htmlFor="teamName"
                >
                  <p
                    className="font-bold text-base"
                    data-testid="label"
                  >
                    Team Name
                  </p>
                  <input
                    className="border-2 w-full rounded p-1 focus:border-gray-500 my-3"
                    data-testid="input"
                    id="teamName"
                    onChange={[Function]}
                    value=""
                  />
                </label>
              </div>
              <div
                className="h-full"
              >
                <label
                  htmlFor="description"
                >
                  <p
                    className="font-bold text-base"
                  >
                    Description
                  </p>
                  <textarea
                    className="resize-none w-full my-3 p-1 border-2 border-gray-400 rounded"
                    data-testid="description"
                    id="description"
                    onChange={[Function]}
                    rows="5"
                    value=""
                  />
                </label>
              </div>
            </div>
            <div
              className="md:w-full md:px-20"
            >
              <div
                className="w-full"
              >
                <label
                  htmlFor="teamWebSite"
                >
                  <p
                    className="font-bold text-base"
                    data-testid="label"
                  >
                    Team website
                  </p>
                  <input
                    className="border-2 w-full rounded p-1 focus:border-gray-500 my-3"
                    data-testid="input"
                    id="teamWebSite"
                    onChange={[Function]}
                    placeholder="https://myteam.com"
                    type="text"
                    value=""
                  />
                </label>
              </div>
              <div
                className=""
              >
                <p
                  className="font-bold text-base mb-2"
                >
                  Team type
                </p>
                <div
                  className="flex justify-between"
                >
                  <button
                    className="flex items-center"
                    data-testid="button-real"
                    onClick={[Function]}
                    type="button"
                  >
                    <div
                      className="rounded-full p-3 border-1 mr-3 relative border-gray-500"
                      data-testid="radio-real"
                    />
                    <p
                      className="inline text-sm text-gray-500"
                      data-testid="label-real"
                    >
                      Real
                    </p>
                  </button>
                  <button
                    className="flex items-center"
                    data-testid="button-fantasy"
                    onClick={[Function]}
                    type="button"
                  >
                    <div
                      className="rounded-full p-3 border-1 mr-3 relative border-gray-500"
                      data-testid="radio-fantasy"
                    />
                    <p
                      className="inline text-sm text-gray-500"
                      data-testid="label-fantasy"
                    >
                      Fantasy
                    </p>
                  </button>
                </div>
              </div>
              <div
                className="my-3 md:h-full"
              >
                <label
                  htmlFor="tags"
                >
                  <p
                    className="font-bold text-base"
                  >
                    Tags
                  </p>
                  <button
                    className="flex rounded border-2 border-gray-400 my-3 py-2 w-full min-height-3 z-10"
                    data-testid="button-field"
                    onClick={[Function]}
                    type="button"
                  >
                    <div
                      className="flex flex-wrap relative"
                      data-testid="tags"
                    >
                      <p
                        className="inline text-base"
                        data-testid="value"
                      >
                        
                      </p>
                    </div>
                    <textarea
                      className="opacity-0 resize-none absolute z-0"
                      data-testid="textarea"
                      id="tags"
                      onChange={[Function]}
                      value=""
                    />
                  </button>
                </label>
              </div>
            </div>
          </div>
        </div>
        <div
          className="mt-5"
        >
          <h1
            className="text-center text-gray-600 text-base"
          >
            CONFIGURE SQUAD
          </h1>
          <div
            className="flex flex-col md:flex-row justify-center py-5 md:px-20"
          >
            <div
              className="md:w-full md:px-20"
            >
              <div>
                <div
                  className="flex items-center mb-3 justify-between md:justify-start"
                >
                  <p
                    className="inline text-base font-bold md:mr-8"
                    label="formation"
                  >
                    Formation
                  </p>
                  <div
                    className="relative w-40"
                  >
                    <button
                      className="bg-white border-2 border-gray-600 rounded cursor-pointer"
                      data-testid="button"
                      onClick={[Function]}
                      type="button"
                    >
                      <div
                        className="flex"
                      >
                        <p
                          className="py-1 w-24"
                          data-testid="value"
                        >
                          Chose one...
                        </p>
                        <div
                          className="border-l-2 border-gray-600 flex items-center justify-center px-1"
                        >
                          <svg
                            className="text-black"
                            fill="currentColor"
                            height={24}
                            size={24}
                            stroke="currentColor"
                            strokeWidth="0"
                            style={
                              Object {
                                "color": undefined,
                              }
                            }
                            viewBox="0 0 24 24"
                            width={24}
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7.41 7.84L12 12.42l4.59-4.58L18 9.25l-6 6-6-6z"
                            />
                          </svg>
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
                <div
                  className="bg-gradient-to-t from-secondary-dark to-secondary-light rounded-sm p-2 mb-5 justify-center"
                  data-testid="formation"
                />
              </div>
              <button
                className="bg-gradient-to-t from-secondary-dark to-secondary-light w-full text-white text-base font-bold rounded-sm py-2 mb-5"
                onClick={[Function]}
                type="button"
              >
                Save
              </button>
            </div>
            <div
              className="md:w-full md:px-20"
            >
              <div
                className="w-full"
              >
                <label
                  htmlFor="searchPlayers"
                >
                  <p
                    className="font-bold text-base"
                    data-testid="label"
                  >
                    Search Players
                  </p>
                  <input
                    className="border-2 w-full rounded p-1 focus:border-gray-500 mb-5"
                    data-testid="search-input"
                    id="searchPlayers"
                    onChange={[Function]}
                    value=""
                  />
                </label>
              </div>
              <div />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="bg-footer py-3 flex justify-center"
    data-testid="footer"
  >
    <p
      className="text-sm text-gray-900"
      data-testid="footer-text"
    >
      2020 - All rights reserved
    </p>
  </div>,
]
`;
